tests/unit/s2n_client_hello_test.c

```diff
@@ -1859,8 +1860,69 @@ int main(int argc, char **argv)
             client_hello.record_version_recorded = 1;
             EXPECT_SUCCESS(s2n_client_hello_get_legacy_record_version(&client_hello, &out));
             EXPECT_EQUAL(out, S2N_TLS12);
-        }
-    }
+        };
+    };
+
+    /* s2n_client_hello_get_server_name() */
+    {
+        /* Safety */
+        {
+            struct s2n_client_hello ch = { 0 };
+            uint16_t length = 0;
+            EXPECT_FAILURE_WITH_ERRNO(s2n_client_hello_get_server_name_length(NULL, &length), S2N_ERR_NULL);
+            EXPECT_FAILURE_WITH_ERRNO(s2n_client_hello_get_server_name_length(&ch, NULL), S2N_ERR_NULL);
+
+            uint8_t buffer;
```

Comment:
nit: Explicit zero-init on the buffer? #4408 and #4404 will automatically deal with this, but for now it's manual.